tests/integration/cli_did_test.go

```diff
@@ -0,0 +1,106 @@
+//go:build integration
+
+package integration
+
+import (
+	"crypto/ed25519"
+
+	"github.com/cheqd/cheqd-node/tests/integration/cli"
+	"github.com/cheqd/cheqd-node/tests/integration/network"
+	"github.com/cheqd/cheqd-node/tests/integration/testdata"
+	cli_types "github.com/cheqd/cheqd-node/x/cheqd/client/cli"
+	"github.com/cheqd/cheqd-node/x/cheqd/types"
+	"github.com/google/uuid"
+	"github.com/multiformats/go-multibase"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("cheqd cli", func() {
+	It("can create diddoc, update it and query the result", func() {
+		// Create a new DID Doc
+		did := "did:cheqd:" + network.DID_NAMESPACE + ":" + uuid.NewString()
+		keyId := did + "#key1"
+
+		pubKey, privKey, err := ed25519.GenerateKey(nil)
+		Expect(err).To(BeNil())
+
+		pubKeyMultibase58, err := multibase.Encode(multibase.Base58BTC, pubKey)
+		Expect(err).To(BeNil())
+
+		payload := types.MsgCreateDidPayload{
+			Id: did,
+			VerificationMethod: []*types.VerificationMethod{
+				{
+					Id:                 keyId,
+					Type:               "Ed25519VerificationKey2020",
+					Controller:         did,
+					PublicKeyMultibase: string(pubKeyMultibase58),
+				},
+			},
+			Authentication: []string{keyId},
+		}
+
+		signInputs := []cli_types.SignInput{
+			{
+				VerificationMethodId: keyId,
+				PrivKey:              privKey,
+			},
+		}
+
+		res, err := cli.CreateDid(payload, signInputs, testdata.BASE_ACCOUNT_1)
+		Expect(err).To(BeNil())
+		Expect(res.Code).To(BeEquivalentTo(0))
+
+		// Update the DID Doc
+		newPubKey, newPrivKey, err := ed25519.GenerateKey(nil)
+		Expect(err).To(BeNil())
+
+		newPubKeyMultibase58, err := multibase.Encode(multibase.Base58BTC, newPubKey)
+		Expect(err).To(BeNil())
+
+		payload2 := types.MsgUpdateDidPayload{
+			Id: did,
+			VerificationMethod: []*types.VerificationMethod{
```

Comment:
I don't see tests done with different kinds of keys, e.g., JsonWebKey2020 vs Ed25519, or trying to use incorrect/unsupported key types (I didn't see an unsupported key type test in negative either)